test/performance/scale_from_zero_test.go

```diff
@@ -57,21 +58,21 @@ func runScaleFromZero(idx int, t *testing.T, clients *test.Clients, ro *test.Res
 		test.DeploymentScaledToZeroFunc,
 		"DeploymentScaledToZero",
 		test.ServingNamespace,
-		2*time.Minute); err != nil {
+		3*time.Minute); err != nil {
 		m := fmt.Sprintf("%d: failed waiting for deployment to scale to zero: %v", idx, err)
 		t.Log(m)
 		return 0, errors.New(m)
 	}
 
 	start := time.Now()
-	t.Log("Waiting for endpoint to serve request")
-	if _, err := pkgTest.WaitForEndpointState(
+	t.Logf("%d: waiting for endpoint to serve request", idx)
+	if _, err := pkgTest.WaitForEndpointStateWithTimeout(
```

Comment:
🤔 It surprises me that this is a `WaitForEndpointState` at all and not a simple check that a curl gets the right thing...  That we need a longer timeout here is even more disturbing 😲 

Do you have a sense for why we are testing this like that?

Reply:
well this is a "curl". But it seems that network programming takes too long during mass attack (503s are being served for quite some time).

Reply:
cc @markusthoemmes 

Still, I think I'd rather see us "Check" here, and test at a lower threshold that we can reliably pass (and then raise that up over time) vs. ignoring the 503s here :(

Reply:
What do you mean by lower threshold here? Fewer instances?

Reply:
I agree, the retries here are unexpected. What form of 503 are you seeing here?

FWIW: The `RetryingRouteInconsistency` should only be used around Route creation, to catch inconsistencies when Istio is propagating things "the first time". One could argue, while annoying, that those errors are not as mission critical. These however are errors returned from a **running** system with no operator interference at all. We should dig into those and avoid them altogether.

Reply:
Well here it is kind of the case. Since there's scale to 0, above, so we basically program route from scratch.
I can try to get the actual 503 printed.

Reply:
The activation path doesn't require route reprogramming, it'd fine for traffic to flow through the activator :(

Reply:
So, I removed 503 from the mix. And it passed at 50 two times.